panther_lights/test/test_led_segment.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdint>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <gtest/gtest.h>
+#include <yaml-cpp/yaml.h>
+
+#include <panther_lights/led_segment.hpp>
+
+class TestLEDSegment : public testing::Test
+{
+public:
+  TestLEDSegment();
+  ~TestLEDSegment() {}
+
+protected:
+  std::shared_ptr<panther_lights::LEDSegment> led_segment_;
+
+  const float controller_freq = 50.0;
+  const std::size_t segment_led_num_ = 10;
+};
+
+TestLEDSegment::TestLEDSegment()
+{
+  auto segment_desc =
+    YAML::Load("{led_range: 0-" + std::to_string(segment_led_num_ - 1) + ", channel: 1}");
+  led_segment_ = std::make_shared<panther_lights::LEDSegment>(segment_desc, controller_freq);
+}
+
+// this method may be moved to utils
+template <typename ExceptionType, typename Func>
+void ExpectThrowWithDescription(const Func & func, const std::string & error)
+{
+  EXPECT_THROW(
+    {
+      try {
+        func();
+      } catch (const ExceptionType & e) {
+        EXPECT_TRUE(std::string(e.what()).find(error) != std::string::npos);
+        throw;
+      }
+    },
+    ExceptionType);
+}
+
+TEST(TestLEDSegmentInitialization, DescriptionMissingRequiredKey)
+{
+  auto segment_desc = YAML::Load("");
```

Comment:
Could we move it to function like CreateSegmentDescription() in test_segment_converter.cpp?